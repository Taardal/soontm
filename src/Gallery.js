import React from "react";
import PropTypes from "prop-types";
import { FlatList, Dimensions } from "react-native";
import GalleryItem from "./GalleryItem";

const numberOfColumns = 3;
const itemSize = Dimensions.get("window").width / numberOfColumns;

class Gallery extends React.Component {
  render() {
    const { movies } = this.props;
    return (
      <FlatList
        data={movies}
        renderItem={({ item }) => (
          <GalleryItem movie={item} width={itemSize} height={itemSize} />
        )}
        keyExtractor={item => item.id}
        numColumns={numberOfColumns}
      />
    );
  }
}

Gallery.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Gallery;
