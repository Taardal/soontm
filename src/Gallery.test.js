import React from "react";
import ReactTestRenderer from "react-test-renderer";
import Gallery from "./Gallery";

jest.mock("FlatList", () => "FlatList");
jest.mock("./GalleryItemContainer", () => "View");

describe("Gallery", () => {
  it("renders expected number of items", () => {
    const movies = [
      { id: 0, value: "movie1" },
      { id: 1, value: "movie2" },
      { id: 2, value: "movie3" },
      { id: 3, value: "movie4" }
    ];

    const viewTree = ReactTestRenderer.create(
      <Gallery
        navigation={{
          navigate: () => jest.fn()
        }}
        movies={movies}
        imageBaseUrl={"imageBaseUrl"}
        imageWidth={100}
        imageHeight={100}
        numberOfColumns={3}
      />
    ).toJSON();
    console.log(viewTree);

    expect(viewTree).toMatchSnapshot();
    expect(viewTree.type).toBe("FlatList");
    expect(viewTree.props.data.length).toBe(movies.length);
  });
});
