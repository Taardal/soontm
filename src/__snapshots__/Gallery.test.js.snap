// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders expected number of items 1`] = `
<FlatList
  data={
    Array [
      Object {
        "id": 0,
        "value": "movie1",
      },
      Object {
        "id": 1,
        "value": "movie2",
      },
      Object {
        "id": 2,
        "value": "movie3",
      },
      Object {
        "id": 3,
        "value": "movie4",
      },
    ]
  }
  keyExtractor={[Function]}
  numColumns={3}
  renderItem={[Function]}
/>
`;
